_posts/2021-01-05-hhvm-4.91.markdown

```diff
@@ -0,0 +1,56 @@
+---
+title: "HHVM 4.91"
+layout: post
+author: jjergus
+category: blog
+---
+
+HHVM 4.91 is released! This release marks the end of support for 4.82;
+HHVM 4.83&ndash;4.88 remain supported, as do the 4.56 and 4.80 LTS releases.
+
+(Note: HHVM 4.85, 4.89 and 4.90 were skipped.)
+
+# Highlights
+
+- The [`dynamic`](https://docs.hhvm.com/hack/built-in-types/dynamic) type can
```

Comment:
Wasn't this already the case? I have been using `<expression> as dynamic` since at least 4.14. I made an [issue](https://github.com/facebook/hhvm/issues/8538) about how `TAny as dynamic` changed its meaning with the introduction of intersection types. The issue itself is not important, but this example code uses `as dynamic`.

```HACK
function from_timestamp(int $timestamp): null {
	$date_time = DateTime::createFromFormat('U', (string)$timestamp) as dynamic;
	invariant($date_time is DateTime, 'Could not turn %d into a DateTime', $timestamp);
	return $date_time;
}
```

Reply:
Hmm good catch, I was looking at https://github.com/facebook/hhvm/commit/deb75cde183fb14831ac4f10548a1a645bbc7342 when I wrote that. Now I'm not sure what it meant by "Update the `as` construct to allow `e as dynamic`." if that already worked :/

Reply:
This commit is talking about 
```
https://github.com/facebook/hhvm/commit/deb75cde183fb14831ac4f10548a1a645bbc7342#diff-846ed6a82ae4dfec1080e4af9d7700e761c331c61ef66d765fe42f88bfaa6a2aR3

Reply:
The commit you reference is talking about
```HACK
class B<T> implements dynamic {
}
```
https://github.com/facebook/hhvm/commit/deb75cde183fb14831ac4f10548a1a645bbc7342#diff-846ed6a82ae4dfec1080e4af9d7700e761c331c61ef66d765fe42f88bfaa6a2aR3

This is in and of itself already new to me.